torch/csrc/profiler/README.md

```diff
@@ -0,0 +1,56 @@
+# Profiler Overview
+
+This README describes the details of how the profiler is implemented.
+
+The profiler instruments PyTorch to collect information about the model's execution. Its main features are:
+* Instrumenting op calls on the CPU side
+* Interfacing with [Kineto](https://github.com/pytorch/kineto/) to collect information from the GPU (or other accelerators)
+* Collecting python stacktraces
+* Exporting this information, e.g. in a chrome trace, or to be processed by downstream tools like [HTA](https://github.com/facebookresearch/HolisticTraceAnalysis)
+
+## Table of Contents
+
+- [Codebase Structure](#codebase-structure)
+- [RecordFunction](#recordfunction)
+- [Autograd Integration](#autograd-integration)
+- [Collection and Post-Processing](#collection-and-post-processing)
+- [Kineto Integration](#kineto-integration)
+- [Python Tracing](#python-tracing)
+
+## Codebase Structure ##
+
+TODO
+
+## RecordFunction ##
+
+[/aten/src/ATen/record_function.h](/aten/src/ATen/record_function.h)
+
+RecordFunction is used by the profiler to instrument CPU-side events.
+
+RecordFunction is a general method of instrumenting function calls in pytorch. It can be used for other general applications, e.g. see [Features for Large-Scale Deployments](https://pytorch.org/docs/stable/notes/large_scale_deployments.html). In PyTorch, it is already included at some important locations; notably, in the [dispatcher](https://github.com/pytorch/pytorch/blob/247c603da9b780534e25fb1d90b6e5a528b625b1/aten/src/ATen/core/dispatch/Dispatcher.h#L650), surrounding every op.
```

Comment:
```suggestion
RecordFunction is a general method of instrumenting function calls in PyTorch. It can be used for other general applications, e.g. see [Features for Large-Scale Deployments](https://pytorch.org/docs/stable/notes/large_scale_deployments.html). In PyTorch, it is already included at some important locations; notably, in the [dispatcher](https://github.com/pytorch/pytorch/blob/247c603da9b780534e25fb1d90b6e5a528b625b1/aten/src/ATen/core/dispatch/Dispatcher.h#L650), surrounding every op.
```